src/soilId/soilIdService.ts

```diff
@@ -17,9 +17,44 @@
 
 import { graphql } from 'terraso-client-shared/graphqlSchema';
 import { SoilIdInputData } from 'terraso-client-shared/graphqlSchema/graphql';
+import { soilDataToIdInput } from 'terraso-client-shared/soilId/soilIdFunctions';
+import { SoilData } from 'terraso-client-shared/soilId/soilIdTypes';
 import * as terrasoApi from 'terraso-client-shared/terrasoApi/api';
 import { Coords } from 'terraso-client-shared/types';
 
+export const fetchSoilMatches = async ({
+  coords,
+  siteId,
+  soilData,
+}: {
+  coords: Coords;
+  siteId?: string;
+  soilData?: SoilData;
+}) => {
+  /*
+   * We call different APIs for different input scenarios.
+   * When soil data is available, we use the data-based soil match API.
+   * When the input is only a location, we use the location-based soil match API.
```

Comment:
@tm-ruxandra @paulschreiber FYI: Oh actually it appears I was misinterpreting the code. It appears that even a site with no data entered will call the data-based one. So it was in fact correct to say the division is Site vs Temporary Location.